programs/stake/src/stake_state.rs

```diff
@@ -1023,6 +1057,34 @@ impl MergeKind {
     }
 }
 
+/// Calculate the effective credits observed for two stakes whenn merging
+///
+/// When merging two activating or active stakes, the credits observed of the
+/// merged stake is the weighted average of the two stakes' credits observed.
```

Comment:
could you add following explanation as well for those who are weak at math?:

```rust
/// This is because we can derive the effective credits_observed by reversing the staking
/// rewards equation, _while keeping the rewards unchanged after merge (i.e. strong
/// requirement)_, like below:
///
/// a(N) => account, r => rewards, s => stake, c => credits:
/// assume:
///   a3 = merge(a1, a2)
/// then:
///   a3.s = a1.s + a2.s
///
/// Next, given:
///   aN.r = aN.c * aN.s (for every N)
/// thus:
///        a3.r = a1.r + a2.r
/// a3.c * a3.s = a1.c * a1.s + a2.c * a2.s
///        a3.c = (a1.c * a1.s + a2.c * a2.s) / (a1.s + a2.s)     // QED
///
/// (For this discussion, we omitted irrelevant variables, including distance
///  calculation against vote_account and point indirection.)


```

Reply:
Absolutely, thanks for this great write-up!